### DISCUSSION TITLE
Wondering whether filtering for "base" game is done right

### DISCUSSION THREAD

**Discussion by The-Gig** [torhus/monster-browser#54] [General]
I was thinking about one thing...
For trying to filter out for the "base" game of a certain game, are we sure that the base game should be managed like the mods?
To recap
- MonsterBrowser's "mod" field from mods.ini file is checked against the server's [game] (usually mod folder name, or not provived) _and_ [gamename] variables.
- In Q3, [gamename] often actually holds the name of the mod, but often it just holds "baseq3" or "Quake3Arena" instead.
- In Openarena, 98% of the servers contains either "baseoa" or "Quake3Arena".
- Some servers have [gamename] set to baseq3/quake3arena/baseoa and [game] set to a mod.
- Servers actually using the base game (baseq3/baseoa) seem to do not provide [game] cvar, right?

So, if the program checks for "baseq3" or "baseoa" the same way it does for "real" mods...
- All the servers which have [gamename] set baseq3 or baseoa are shown, including those where [game] is set to something (which may be, as example, osp), which means we are showing some modded servers to people who are interested into unmodded servers.
- All the servers which have [gamename] set to Quake3Arena are not shown, despite in reality being running the basegame.

So, wondering how it could be improved... Maybe there could be a way to _check for missing [game] cvar_, instead? But in mods.ini you already have blank "mod=" as "no filter" and missing "mod" line as "same as section name".

Further notes:
- I considered Quake3 and OpenArena. Other games may differ...
- [game] and [gamename] cvars seem to be of the "info" kind, while of cvars of "status" kind [game] doesn't exist and [gamename] may have a different value (see this <https://github.com/torhus/monster-browser/issues/4#issuecomment-1296935757>).

**Comment by The-Gig** [comment]
Any thoughts about this topic?

**Reply by torhus** [reply]
Not yet, probably needs more advanced filtering.
